File: app/src/main/java/com/phdlabs/sungwon/a8chat_android/api/utility/GsonHolder.java
package com.phdlabs.sungwon.a8chat_android.api.utility;

import com.google.gson.Gson;

/**
 * Created by SungWon on 9/18/2017.
 */

public class GsonHolder {
    private static GsonHolder sInstance;

    private Gson gson;
    public static GsonHolder getInstance() {
        if (sInstance == null) {
            sInstance = new GsonHolder();
        }
        return sInstance;
    }

    private GsonHolder() {
        gson = new Gson();
    }

    public static Gson get() {
        return getInstance().gson;
    }
}

File: app/src/main/java/com/phdlabs/sungwon/a8chat_android/api/utility/GsonHolder.kt
package com.phdlabs.sungwon.a8chat_android.api.utility

import com.google.gson.Gson

/**
 * Created by JPAM on 12/20/17.
 * Gson Singleton
 */
class GsonHolder {

    /*Singleton*/
    companion object {
        var instance: GsonHolder = GsonHolder()
            private set
    }

    /*Properties*/
    private var gson: Gson? = null

    /*Initialization*/
    init {
        gson = Gson()
    }

    /*Getter*/
    fun get(): Gson? = gson
}